examples/schema.rs

```diff
@@ -0,0 +1,137 @@
+#[macro_use]
+extern crate clap;
+use clap::{App, ArgMatches};
+use ion_rs::text::writer::TextWriter;
+use ion_rs::value::reader::element_reader;
+use ion_rs::value::reader::ElementReader;
+use ion_rs::value::writer::{ElementWriter, Format, TextKind};
+use ion_rs::IonType;
+use ion_schema::authority::{DocumentAuthority, FileSystemDocumentAuthority};
+use ion_schema::result::IonSchemaResult;
+use ion_schema::system::SchemaSystem;
+use std::fs;
+use std::path::Path;
+use std::str::from_utf8;
+
+fn main() -> IonSchemaResult<()> {
+    // load the YAML file which has all the CLI commands information stored in it
+    let yaml = load_yaml!("cli.yaml");
+    let matches = App::from_yaml(yaml).get_matches();
+    let (command_name, command_args) = matches.subcommand();
+    let command_args = command_args.unwrap();
+
+    match command_name {
+        "load" => load(command_args)?,
+        "validate" => validate(command_args)?,
+        _ => eprintln!("command name: {} not found", command_name),
+    }
+
+    Ok(())
+}
+
+fn load(command_args: &ArgMatches) -> IonSchemaResult<()> {
+    // Extract the user provided authorities
+    let authorities: Vec<_> = command_args.values_of("directories").unwrap().collect();
+
+    // Extract schema file provided by user
+    let schema_id = command_args.value_of("schema").unwrap();
+
+    // Set up authorities vector
+    let mut document_authorities: Vec<Box<dyn DocumentAuthority>> = vec![];
+
+    for authority in authorities {
+        document_authorities.push(Box::new(FileSystemDocumentAuthority::new(Path::new(
+            authority,
+        ))))
+    }
+
+    // Create a new schema system from given document authorities
+    let mut schema_system = SchemaSystem::new(document_authorities);
+
+    // load schema
+    let schema = schema_system.load_schema(schema_id);
+
+    if schema.is_ok() {
+        eprintln!("Schema: {:?} was successfully loaded", schema.unwrap().id());
+    } else {
+        eprintln!("{:?}", schema.unwrap_err());
+    }
+
+    Ok(())
```

Comment:
Since `main` returns an `IonSchemaResult`, it seems like you could replace this with:

```rust
eprintln!("Schema: {:?} was successfully loaded",  schema_system.load_schema(load_id)?);
Ok(())
```
?